src/main/java/com/google/sps/data/SessionContext.java

```diff
@@ -56,4 +71,19 @@ public String getLoggedInUserId() {
   public boolean isUserLoggedIn() {
     return userService.isUserLoggedIn();
   }
+
+  public boolean userProfileExists() {
+    String id = getLoggedInUserId();
+    if(id != null) {
+        try {
+            User userExists = PersistentUserRepository.getInstance().fetchUserWithId(id);
+            return userExists.getId().equals(id);
```

Comment:
[L80] Given fetchUserWithId will either throw a failure or return the user with that id, wouldn't `userExists.getId().equals(id)` always be true?

Given you're now relying on fetchUserWithId to also check if a user exists, it probably makes sense to make `fetchUserWithId` return null if it doesn't find anything, rather than throwing (PersistentUserRepository.java:185).

Throwing/exceptions is generally reserved for failure conditions; They're expensive to process and communicate to the reader that what they address is unexpected behavior. Relying on them as part of ordinary operations is usually discouraged.

Could you change `fetchUserWithId` so it returns null if a value isn't found, and then process the null/not null cases here (instead of the try-catch). Alternatively, a cleaner solution would be to add a 'hasUserWithId' to the UserRepository interfact that avoids fetching a full user from the DB (expensive) and instead just does a check.

Reply:
Done

Reply:
Could you address the first part, regarding:

"
Given fetchUserWithId will either throw a failure or return the user with that id, wouldn't userExists.getId().equals(id) always be true?
"

Reply:
I added a condiitonal check for when its not null saying
`
if(userExists != null){
  return userExists.getId().equals(id);
}`

Reply:
Fixed it. Done